include/revng/Pypeline/MapContainer.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Pypeline/Common.h"
+#include "revng/Pypeline/ObjectID.h"
+
+namespace revng::pypeline {
+
+namespace detail {
+
+template<ObjectID::Kind KindParam, ConstexprString NameParam>
+class MapContainer {
+public:
+  static constexpr llvm::StringRef Name = NameParam;
+  static constexpr ObjectID::Kind Kind = KindParam;
+
+private:
+  std::map<ObjectID, Buffer> Map;
+
+public:
+  std::set<ObjectID> objects() const {
+    return std::views::keys(Map) | revng::to<std::set<ObjectID>>();
+  }
```

Comment:
What if we avoid materializing and return the view?
Doable?

Reply:
I don't know enough about views to say that it would work.
Also it would be pretty annoying to handle on the nanobind side because the caster wouldn't know what to convert the view into (was it `list`, `set` or `dict`?)